apps/extension/src/routes/popup/home/index.tsx

```diff
@@ -24,9 +30,19 @@ export const popupIndexLoader = async (): Promise<Response | PopupLoaderData> =>
   };
 };
 
+const getAddrByIndex =
+  (wallet?: Wallet) =>
+  (index: number, ephemeral: boolean): Address => {
+    if (!wallet) throw new Error('No active wallet');
+
+    const fullViewingKey = FullViewingKey.fromJsonString(wallet.fullViewingKey);
+    return ephemeral
+      ? getEphemeralByIndex(fullViewingKey, index)
+      : getAddressByIndex(fullViewingKey, index);
+  };
+
```

Comment:
Somewhat related to linting, but wanted to remove the exhaustive deps lint exclusion from `packages/ui/components/ui/select-account.tsx`. Moved this from zustand to here. By changing this function to take a wallet (instead of all slices) we don't have the constant re-rendering issue that messed with ephemeral addresses like before.